main.go

```diff
@@ -48,67 +48,67 @@ func main() {
 		esTimeout = kingpin.Flag("es.timeout",
 			"Timeout for trying to get stats from Elasticsearch.").
-			Default("5s").Envar("ES_TIMEOUT").Duration()
+			Default("5s").Duration()
 		esAllNodes = kingpin.Flag("es.all",
 			"Export stats for all nodes in the cluster. If used, this flag will override the flag es.node.").
-			Default("false").Envar("ES_ALL").Bool()
+			Default("false").Bool()
 		esNode = kingpin.Flag("es.node",
 			"Node's name of which metrics should be exposed.").
-			Default("_local").Envar("ES_NODE").String()
+			Default("_local").String()
 		esExportIndices = kingpin.Flag("es.indices",
 			"Export stats for indices in the cluster.").
-			Default("false").Envar("ES_INDICES").Bool()
+			Default("false").Bool()
 		esExportIndicesSettings = kingpin.Flag("es.indices_settings",
 			"Export stats for settings of all indices of the cluster.").
-			Default("false").Envar("ES_INDICES_SETTINGS").Bool()
+			Default("false").Bool()
 		esExportIndicesMappings = kingpin.Flag("es.indices_mappings",
 			"Export stats for mappings of all indices of the cluster.").
-			Default("false").Envar("ES_INDICES_MAPPINGS").Bool()
+			Default("false").Bool()
 		esExportClusterSettings = kingpin.Flag("es.cluster_settings",
 			"Export stats for cluster settings.").
-			Default("false").Envar("ES_CLUSTER_SETTINGS").Bool()
+			Default("false").Bool()
 		esExportShards = kingpin.Flag("es.shards",
 			"Export stats for shards in the cluster (implies --es.indices).").
-			Default("false").Envar("ES_SHARDS").Bool()
+			Default("false").Bool()
 		esExportSnapshots = kingpin.Flag("es.snapshots",
 			"Export stats for the cluster snapshots.").
-			Default("false").Envar("ES_SNAPSHOTS").Bool()
+			Default("false").Bool()
 		esClusterInfoInterval = kingpin.Flag("es.clusterinfo.interval",
 			"Cluster info update interval for the cluster label").
-			Default("5m").Envar("ES_CLUSTERINFO_INTERVAL").Duration()
+			Default("5m").Duration()
 		esCA = kingpin.Flag("es.ca",
 			"Path to PEM file that contains trusted Certificate Authorities for the Elasticsearch connection.").
-			Default("").Envar("ES_CA").String()
+			Default("").String()
 		esClientPrivateKey = kingpin.Flag("es.client-private-key",
 			"Path to PEM file that contains the private key for client auth when connecting to Elasticsearch.").
-			Default("").Envar("ES_CLIENT_PRIVATE_KEY").String()
+			Default("").String()
 		esClientCert = kingpin.Flag("es.client-cert",
 			"Path to PEM file that contains the corresponding cert for the private key to connect to Elasticsearch.").
-			Default("").Envar("ES_CLIENT_CERT").String()
+			Default("").String()
 		esInsecureSkipVerify = kingpin.Flag("es.ssl-skip-verify",
 			"Skip SSL verification when connecting to Elasticsearch.").
-			Default("false").Envar("ES_SSL_SKIP_VERIFY").Bool()
+			Default("false").Bool()
 		esApiKey = kingpin.Flag("es.apiKey",
 			"API Key to use for authenticating against Elasticsearch").
-			Default("").Envar("ES_API_KEY").String()
```

Comment:
I think this one should stay because it's sensitive.

Reply:
Good point, fixed.

Reply:
Actually, we should probably remove this flag.

Reply:
This was added recently: https://github.com/prometheus-community/elasticsearch_exporter/pull/459
It supports elastic cloud instances that use the API key for auth. Are you suggesting an alternative or that it is not necessary at all?